### PR TITLE
add distance method

### PR DESCRIPTION
**Pull request by ghost**
note use of approx in testing

### CODE REVIEW

File: math_utils.cpp [hunk 1]
```diff
@@ -77,3 +77,17 @@ bool Math::EqualParity(std::vector<int> nums) {
   // Return true if no parity mismatches were found
   return true;
 }
+
+
+/*
+Calulates distance between two points of arbitrary dimension
```
Comment (kateosborn10) [inline math_utils.cpp:83]: Function comment does not adhere to style guidelines: needs  to have an @param and @return descriptors

File: math_utils.cpp [hunk 2]
```diff
@@ -77,3 +77,17 @@ bool Math::EqualParity(std::vector<int> nums) {
...
+*/
+double Math::distance(std::vector<int> p1, std::vector<int> p2) {
```
Comment (kateosborn10) [inline math_utils.cpp:85]: method names should be capitalized: should be Math::Distance()

File: test.cpp [hunk 3]
```diff
@@ -11,8 +11,19 @@ Tests for the Math Class
 #include "math_utils.h"
 #include <iostream>
 #include <vector>
+#include <limits>
 
 
+TEST_CASE ("Testing distance") {
```
Comment (kateosborn10) [inline test.cpp:17]: The following tests would be easier to read if they were broken up into two sections. You have two require clauses that are testing different cases. The first require is testing the case where the two points are equal to each other and the second require is testing two points that are not equal to each other.

File: math_utils.cpp [hunk 4]
```diff
@@ -77,3 +77,17 @@ bool Math::EqualParity(std::vector<int> nums) {
...
+   if (p1.size() != p2.size()) throw std::invalid_argument("Invalid input!");
```
Comment (kateosborn10) [inline math_utils.cpp:86]: The throw statement should be on the next line and indented according to the style guides.

File: math_utils.h [hunk 5]
```diff
@@ -10,6 +10,8 @@ class Math {
   static int GetDigit(int num, int place);
   static bool EqualParity(int x, int y);
   static bool EqualParity(std::vector<int> nums);
+
+    static double distance(std::vector<int> p1, std::vector<int> p2);
```
Comment (kateosborn10) [inline math_utils.h:14]: This line should match the indentations of the lines above it.

**Review by arnabp — CHANGES_REQUESTED** [review]
Math::distance function comment has a minor typo

Function comment should have @param and @return descriptions, to match code style.

"distance" function should be named "Distance" to match PascalCase.

File: test.cpp [hunk 6]
```diff
@@ -11,8 +11,19 @@ Tests for the Math Class
...
+    std::vector<int> p1 = {1, 2, 3};
+    std::vector<int> p2 = {1, 2, 3};
+    REQUIRE(Math::distance(p1, p2) == 0);
+    std::vector<int> p3 = {1, 1, 1};
+    double ans = 2.2360679775 + std::numeric_limits<double>::epsilon();
```
Comment (kateosborn10) [inline test.cpp:22]: Test cases for your method should be exhaustive. Try adding sections that test points in 1D, 2D, etc. Also test for the corner cases, such as when the point vectors are empty or the size of the point vectors don't match.

**Review by kateosborn10 — CHANGES_REQUESTED** [review]
Some minor changes are needed for this feature to match the design and style guidelines used in the rest of the repo. Additional tests are also required.
